numpy/random/tests/test_generator_mt19937.py

```diff
@@ -13,6 +14,26 @@
 
 random = Generator(MT19937())
 
+JUMP_TEST_DATA = [
```

Comment:
I'd probably add another test covering the issue I originally opened:

```py
    { # You already have this one
        "seed": 0,
        "steps": 10,
        "initial": {"key_md5": "64eaf265d2203179fb5ffb73380cd589", "pos":  9},
        "jumped": {"key_md5": "8cb7b061136efceef5217a9ce2cc9a5a", "pos": 598},
    },
    { # But don't have this one
        "seed": 0,
        "steps": 11,
        "initial": {"key_md5": "64eaf265d2203179fb5ffb73380cd589", "pos": 10},
        "jumped": {"key_md5": "8cb7b061136efceef5217a9ce2cc9a5a", "pos": 599},
    },
```

The added test should ensure that we have the following behaviour:
`jump(next(mt19937)) == next(jump(mt19937))` - _or something close_

Reply:
The only guarantee that I think we should try and make is that the code is faithful to the original author's code.  Whether this is true or not I think is a different issue.

Reply:
If I understand it correctly, the test you propose will not work (in the current implementation / original authors source code:

> `jump(next(mt19937)) == next(jump(mt19937))` - _or something close_

However, if this is the case, then I don't think that this pull request closes  #15394, does it?
Or put it the other way around: With the current state of the PR, shouldn't the [documentation](https://numpy.org/doc/stable/reference/random/bit_generators/generated/numpy.random.MT19937.jumped.html?highlight=numpy%20random%20mt19937%20mt19937%20jumped#numpy.random.MT19937.jumped) be adopted too?

> The state of the returned big generator is jumped as-if 2**(128 * jumps) random numbers have been generated.

Reply:
Another note: Shall we implement the same type of test

> `jump(next(mt19937)) == next(jump(mt19937))` - _or something close_

in the tests for  `PCG64`? There it should pass; but nice to have this ensured for the future, too.

Reply:
Depends on what the point of #15394 is.  If it is to check the implementation, then I think this PR closes that.  If it is to make jumped work in a way different from the original code, then it does not.  I don't think the latter is realistic.

Reply:
My original point when opening the issue was the following:

As a user of the library I saw that the jump was supposed to offset by a fixed number of iterations. So my expectation and the reason why I opened the issue was `jump(next(mt19937))` should always be the same as `next(jump(mt19937))`. If it is not the case it seems problematic as it looks like the `2**128` skipped iterations are not true so I cannot safely rely on a jump if we don't know where it jumps.

Actually I found this issue while implementing a jump for mersenne twister on a [JavaScript library I am maintaining](https://github.com/dubzzz/pure-rand/pull/12). I already had a working implementation of the mersenne generator but needed a reference one for the jump to compare my results to it. So I compared it to the jump of numpy. In the set of tests, I use on my generators [one of them](https://github.com/dubzzz/pure-rand/blob/d3fa7faa517c13fd4315f4135de91179dde2f02d/test/unit/generator/RandomGenerator.properties.ts#L34-L43) - _based on property based testing (see [hypothesis](https://hypothesis.works/) for Python)_ - checks that: `jump(next(generator)) == next(jump(generator))`. And here is how I found the problem.